mdn/src/main/scala/org/apache/james/mdn/MDNReportParser.scala

```diff
@@ -0,0 +1,475 @@
+                                                                              dispositionType: DispositionType,
+                                                                              dispositionModifiers: Option[Seq[DispositionModifier]]) =>
+       Disposition.builder()
+         .actionMode(modes._1)
+         .sendingMode(modes._2)
+         .`type`(dispositionType)
+         .addModifiers(dispositionModifiers.getOrElse(Nil):_*)
+         .build()
+      )
+  }
+
+
+
+  //    disposition-mode = action-mode OWS "/" OWS sending-mode
+  private def dispositionMode: Rule1[(DispositionActionMode, DispositionSendingMode)] = rule {
+    (capture(actionMode) ~ ows ~ "/" ~ ows ~ capture(sendingMode)) ~> ((actionMode: String, sendingMode: String) => {
+      val action = actionMode match {
+        case "manual-action" => DispositionActionMode.Manual
+        case "automatic-action" => DispositionActionMode.Automatic
+      }
+      val sending = sendingMode match {
+        case "MDN-sent-manually" => DispositionSendingMode.Manual
+        case "MDN-sent-automatically" => DispositionSendingMode.Automatic
+      }
+      (action, sending)
+    })
+  }
+
+  //    action-mode = "manual-action" / "automatic-action"
+  private def actionMode = rule { "manual-action" | "automatic-action" }
+
+  //    sending-mode = "MDN-sent-manually" / "MDN-sent-automatically"
+  private def sendingMode = rule {"MDN-sent-manually" | "MDN-sent-automatically" }
+
+  /*    disposition-type = "displayed" / "deleted" / "dispatched" /
+                      "processed"    */
+  private def dispositionType : Rule1[DispositionType] = rule {
+    "displayed" ~ push(DispositionType.Displayed) |
+    "deleted" ~ push(DispositionType.Deleted) |
+    "dispatched" ~ push(DispositionType.Dispatched) |
+    "processed" ~ push(DispositionType.Processed)
+  }
+
+  private def dispositionModifiers: Rule1[Seq[DispositionModifier]] = rule { (ows ~ "/" ~ ows ~ capture(dispositionModifier) ~
+      zeroOrMore(ows ~ "," ~ ows ~ capture(dispositionModifier))) ~> ((head: String, tail: Seq[String]) =>
+      tail.prepended(head).map(new DispositionModifier(_)))
+    }
+
+
+  //    disposition-modifier = "error" / disposition-modifier-extension
+  private def dispositionModifier = rule { "error" | dispositionModifierExtension }
+
+  //    disposition-modifier-extension = Atom
+  private def dispositionModifierExtension = rule { atom }
+
+  //    error-field = "Error" ":" *([FWS] text)
+  private[mdn] def errorField: Rule1[Error] = rule { ("Error" ~ ":" ~ capture(zeroOrMore(optional(fws) ~ text))) ~> ((error: String) =>  new Error(Text.fromRawText(error))) }
+
+  //    extension-field = extension-field-name ":" *([FWS] text)
+  private[mdn] def extentionField: Rule1[ExtensionField] = rule { capture(extensionFieldName) ~ ":" ~ capture(zeroOrMore(optional(fws) ~ text)) ~> ((extensionFieldName: String, text : String) =>
```

Comment:
do you really need to capture optional `fws` before `text`?

Reply:
I think yes, as i understand it  the extension-field-name value could be like : 
`part1 part2 partn`